ciphers/XOR_cipher.py

```diff
@@ -0,0 +1,239 @@
+		"""
+			input: 'content' of type string and 'key' of type int
+			output: encrypted string 'content' as a list of chars
+			if key not passed the method uses the key by the constructor.
+			otherwise key = 1
+		"""
+
+		# precondition
+		assert (isinstance(key,int) and isinstance(content,str))
+
+		# testing for default arguments
+		if (key == 0):
+			if (self.__key == 0):
+				key = 1
+			else:
+				key = self.__key
+
+		# make sure key can be any size
+		while (key > 255):
+			key -= 255
+
+		# This will be returned
+		ans = []
+		resultNumber = 0
+
+		for ch in content:
+			ans.append(chr(ord(ch) ^ key))
+
+		return ans
+
+	def decrypt(self,content,key):
+		"""
+			input: 'content' of type list and 'key' of type int
+			output: decrypted string 'content' as a list of chars
+			if key not passed the method uses the key by the constructor.
+			otherwise key = 1
+		"""
+
+		# precondition
+		assert (isinstance(key,int) and isinstance(content,list))
+
+		# testing for default arguments
+		if (key == 0):
+			if (self.__key == 0):
+				key = 1
+			else:
+				key = self.__key
+
+		# make sure key can be any size
+		while (key > 255):
+			key -= 255
+
+		# This will be returned
+		ans = []
+		resultNumber = 0
+
+		for ch in content:
+			ans.append(chr(ord(ch) ^ key))
+
+		return ans
+
+
+	def encrypt_string(self,content, key = 0):
+		"""
+			input: 'content' of type string and 'key' of type int
+			output: encrypted string 'content'
+			if key not passed the method uses the key by the constructor.
+			otherwise key = 1
+		"""
+
+		# precondition
+		assert (isinstance(key,int) and isinstance(content,str))
+
+		# testing for default arguments
+		if (key == 0):
+			if (self.__key == 0):
+				key = 1
+			else:
+				key = self.__key
+
+		# make sure key can be any size
+		while (key > 255):
+			key -= 255
+
+		# This will be returned
+		ans = ""
+		resultNumber = 0
+
+		for ch in content:
+			ans += chr(ord(ch) ^ key)
+
+		return ans
+
+	def decrypt_string(self,content,key = 0):
+		"""
+			input: 'content' of type string and 'key' of type int
+			output: decrypted string 'content'
+			if key not passed the method uses the key by the constructor.
+			otherwise key = 1
+		"""
+
+		# precondition
+		assert (isinstance(key,int) and isinstance(content,str))
+
+		# testing for default arguments
+		if (key == 0):
+			if (self.__key == 0):
+				key = 1
+			else:
+				key = self.__key
+
+		# make sure key can be any size
+		while (key > 255):
+			key -= 255
+
+		# This will be returned
+		ans = ""
+		resultNumber = 0
+
+		for ch in content:
+			ans += chr(ord(ch) ^ key)
+
+		return ans
+
+
+	def encrypt_file(self, file, key = 0):
+		"""
+			input: filename (str) and a key (int)
+			output: returns true if encrypt process was
+			successful otherwise false
+			if key not passed the method uses the key by the constructor.
+			otherwise key = 1
+		"""
+
+		#precondition
+		assert (isinstance(file,str) and isinstance(key,int))
+
+		try:
+			fin = open(file,"r")
```

Comment:
use __with open()__ to avoid the need to close() manually and also to close() even if an exception is thrown.